openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/SWIGTrainData.java

```diff
@@ -134,6 +182,15 @@ public void initSwigTrainLabelDataArray() {
         this.swigLabelsChunkedArray.coalesce_to(this.swigTrainLabelDataArray);
     }
 
+    /**
+     * Initializes the swigTrainLabelDataArray and copies
+     * the chunked labels array data to it.
+     */
+    public void initSwigConstraintGroupDataArray() {
```

Comment:
Looking back at the code I think it's easiest to delete the chunked data as soon as it has moved to the coalesced representation. We should rename the method and add the last line with `release()`, which is the "delete()" for chunked arrays:

```suggestion
    public void coalesceSwigConstraintGroupDataChunkedArray() {
        this.swigConstraintGroupDataArray = lightgbmlib.new_floatArray(this.swigConstraintGroupChunkedArray.get_add_count());
        this.swigConstraintGroupChunkedArray.coalesce_to(this.swigConstraintGroupDataArray);
        this.swigConstraintGroupDataArray.release(); // Release as soon as it's not needed.
    }
```

We should do the same on the method above, `initSwigTrainLabelDataArray` -> `coalesceSwigTrainLabelDataChunkedArray` and call `release()` at the end.

Reply:
@mlobofeedzai FIY we're changing this to avoid memory leaks and have a simpler code

Reply:
Yes that seems better. Release all chunked arrays as soon as we coalesce them to the data arrays.

Minor point: suggestion above should release the chunked array right?
`this.swigConstraintGroupChunkedArray.release()`

Reply:
Exactly @AndreFCruz  :) My bad ;)

Reply:
done :slightly_smiling_face: